### ISSUE TITLE
Save state cannot be loaded

### ISSUE THREAD

**Issue by thecodemonkey86** [project64/project64#2405]
### Expected behavior?

application should not freeze on loading save state and/or save state should be written correctly if it should be corrupted (I don't know if that's the case)

### Actual behavior?

application freezes, only closing is still possible

### Steps to reproduce the problem

1. start game Diddy Kong Racing (U) (V1.1) [!]
2. try loading state (see attachment)
[Diddy Kong Racing-40D835036C4C7517178B7762F82E9E41.zip](https://github.com/project64/project64/files/14074404/Diddy.Kong.Racing-40D835036C4C7517178B7762F82E9E41.zip)

### Error messages?

(no error messages / log entries)

### What game were you playing?

Diddy Kong Racing (U) (V1.1) [!]

### Specifications and build details

Project64: 64-bit, 71067ccd

OS: Windows 10 22H2 Build 19045.3930

CPU: Intel(R) Core(TM) i7-4770 CPU 

RAM: 16 GB

Graphics: NVIDIA GeForce GTX 760

### Using included plugins

- [X] I can confirm I am using the included RSP, audio, input, and graphics plugins

### Plugins used while the issue occurred

Graphics plugin (and HLE or LLE): Project64 Video Plugin (HLE)
Audio plugin: Project64  audio plugin
Input plugin: N-Rage
RSP plugin: Project64 RSP plugin

### Copyright Compliance

- [X] There are no infringing files attached to this bug report.

**Comment by Squall-Leonhart** [comment]
Save state is bad, there is no fixing it.

**Comment by project64** [comment]
if the save state is working correctly on the current stable build but failing on the latest there is something I can do.

if you are generating a save in a particular way, then reloading it and it fails then I can do something about it.

When a particular save is failing to load, I have no clue if it is loading it that is the issue or something has corrupted the save state. If the save state is corrupted then I can not fix things with the save state. i need to know how to manually generate the save state in the first place

**Comment by thecodemonkey86** [comment]
- it cannot be loaded on the current stable version (also valid for "good" save states that can be loaded in the dev version, but maybe that's because of 64-bit?)
> generating a save in a particular way
- I don't really know what you mean by that or how one would do that, I just pressed F5 (using the same version I try to load it with, https://github.com/project64/project64/commit/71067ccdc4194ff6993134632d04a88363fb3757)
- it's not important that this particular save state cannot be fixed, just wanted to inform that the issue exists

**Comment by Squall-Leonhart** [comment]
The current master can load states that it has made, as well as states made by PJ64 1.6 back in 2006.

**Comment by thecodemonkey86** [comment]
OK, but vice versa does not work for me

**Comment by Squall-Leonhart** [comment]
Your state was broken because of https://github.com/project64/project64/commit/15175d3fe25a642c5fee32229153dc96f876f0e6

it cannot work on any version of pj64 (and states made on master shouldn't be expected to work on 3.x or 2.x or 1.x either)

**Comment by thecodemonkey86** [comment]
The latter is totally fine, but isn't revision https://github.com/project64/project64/commit/15175d3fe25a642c5fee32229153dc96f876f0e6 older than https://github.com/project64/project64/commit/71067ccdc4194ff6993134632d04a88363fb3757, so that this fix should be already included in the version I'm using?

**Comment by Squall-Leonhart** [comment]
there is clear and apparent differences to the hex contents of the state you have made and both the ones i have that work  that work against your claim that you made it on 71067cc.

There is additional garbage in the header of yours, as well as data missing at other points in mine - So if you made this on 71067cc, then its best you start testing with memtest and prime95, because you have hardware corruption occuring.

**Comment by thecodemonkey86** [comment]
I ran memtest and other hardware/HDD tests without any errors. My hardware is fine

**Comment by DerekTurtleRoe** [comment]
@thecodemonkey86 Just to be clear, this save state loads fine in the version you made it with, but not any newer development builds using the same plugin versions?

If so that kind of makes sense, since a lot of changes to accuracy have occurred in development builds.

Also did you try loading it with the CPU interpreter and RSP interpreter?

If you did any Windows updates between versions that might be relevant as well.

**Comment by thecodemonkey86** [comment]
> this save state loads fine in the version you made it with

no, I never said that. The save state was created using a nightly 64-bit build based on https://github.com/project64/project64/commit/71067ccdc4194ff6993134632d04a88363fb3757, but it got corrupted, and cannot be loaded in any version. On my system no change of PJ64 version was made in this particular case nor any other changes

**Comment by DerekTurtleRoe** [comment]
> > this save state loads fine in the version you made it with
> 
> no, I never said that. The save state was created using a nightly 64-bit build based on [71067cc](https://github.com/project64/project64/commit/71067ccdc4194ff6993134632d04a88363fb3757), but it got corrupted, and cannot be loaded in any version. On my system no change of PJ64 version was made in this particular case nor any other changes

Oh, I see. If you didn't change anything that is odd, although I can't seem to reproduce the issue on my end, which is unfortunate.

**Comment by postalesdeamor** [comment]
Just download Project64-Win32-Dev-4.0.0-6469-c39582b.

I started the game "Mega Man 64", saved the state. Then I try to load the state and the error appears: "CN64System:EmulationStarting: Exception caught File: N64System\N64System.cpp Line: 736".

Project64.log file:

"2024/10/19 04:36:44.319 01968,Error,User Interface,CNotificationImp::DisplayError,CN64System::EmulationStarting: Exception caught
File: N64System\N64System.cpp
Line: 736"
